flight/Modules/Airspeed/airspeed.c

```diff
@@ -104,10 +104,10 @@ static int8_t airspeedADCPin=-1;
 // Private functions
 static void airspeedTask(void *parameters);
 void baro_airspeedGet(BaroAirspeedData *baroAirspeedData, uint32_t *lastSysTime, uint8_t airspeedSensorType, int8_t airspeedADCPin);
-static void AirspeedSettingsUpdatedCb(UAVObjEvent * ev);
+static void AirspeedSettingsUpdatedCb(UAVObjEvent * ev, void *ctx, void *obj, int len);
```

Comment:
1 is a great argument.  There's enough to verify and correct on this path of code change already without incurring a ton more risk.
2 is important for callback efficiency.  Most likely use of this parameter is to be assigned to a local variable struct my_state *foo = ctx;    Then do stuff with foo->whatever.  The compiler will elide the assignment and then access to foo->whatever is just an indirect load.  If it's a structure pointer, then every time you escape the optimizer's analysis by calling into another module you need to load, load, load again. 

It also lets us use things like restrict, which we can't do meaningfully with structures being passed in.  These things make a big difference on performance on giant machines like I'm used to writing code for; it's even bigger down here.

Basically, I don't see how

```
struct foo { .ctx=blah, .obj=obj, .len=len, .event=ev } bar;
cb(&bar);
```

is preferable to `cb(ev, blah, obj, len);`, and it's obviously less efficient on several axes and wordier.